src/peft/config.py

```diff
@@ -63,7 +90,7 @@ def save_pretrained(self, save_directory: str, **kwargs) -> None:
         os.makedirs(save_directory, exist_ok=True)
         auto_mapping_dict = kwargs.pop("auto_mapping_dict", None)
 
-        output_dict = asdict(self)
+        output_dict = self.to_dict()
```

Comment:
I originally modified it because the other tests (the ones requiring that `load_from_json()` and `to_dict()` were identical) were now failing due to the extra `runtime` dictionary in the latter. I think moving the runtime stuff to the `LoraConfig` will address this partially, although I think those tests will now fail for `LoraConfig` only instead.

Reply:
It's okay if the LoRA unit test would need to be changed to account for this new key.

Reply:
Agreed with @BenjaminBossan with the same reasons he stated, if you could update the failing tests accordingly it would be really great

Reply:
Reverted change and dropping runtime key from `output_dict` instead.

Reply:
A tiny sidenote: I overrode this method in `LoraConfig` and dropped runtime there. Keeping `output_dict = self.to_dict()` here means no other changes necessary, so I kept that.